pinot-common/src/main/java/org/apache/pinot/common/function/scalar/MapFunctions.java

```diff
@@ -0,0 +1,213 @@
+        List<Map.Entry<Object, Object>> keyList = new ArrayList<>(convertedMap.size());
+        keyList.addAll(convertedMap.keySet());
+
+        // convert object list to string list
+        List<String> stringKeysList = new ArrayList<String>(convertedMap.size());
+        for (Object key : keyList) {
+            stringKeysList.add(key.toString());
+        }
+
+        // convert string keys list to string array and return
+        return Arrays.copyOf(stringKeysList.toArray(), stringKeysList.size(), String[].class);
+    }
+
+    /**
+     * Scalar listValues UDF that lists all values from a String representation of JSON Map
+     *
+     * @param jsonString
+     * @return string array of all values from JSON map in data
+     */
+    @ScalarFunction
+    public static String[] listValues(String jsonString) {
+        // handle empty string case
+        if (jsonString == null) {
+            return new String[0];
+        }
+
+        // convert string input to equivalent java map
+        LinkedHashMap convertedMap = (LinkedHashMap) stringToMap(jsonString);
+
+        // add map key elements to list
+        List<Map.Entry<Object, Object>> keyList = new ArrayList<>(convertedMap.size());
+        keyList.addAll(convertedMap.keySet());
+
+        // convert object list to string list
+        List<String> stringValuesList = new ArrayList<String>(convertedMap.size());
+        for (Object key : keyList) {
+            if (convertedMap.get(key) == null) {
+                stringValuesList.add("null");
+                continue;
+            }
+            stringValuesList.add(convertedMap.get(key).toString());
+        }
+
+        // convert string keys list to string array and return
+        return Arrays.copyOf(stringValuesList.toArray(), stringValuesList.size(), String[].class);
+    }
+
+    /**
+     * Scalar mapSize UDF that returns size/number of entries of the Map represented by a JSON Map
+     * String
+     *
+     * @param jsonString
+     * @return size of JSON map in data
+     */
+    @ScalarFunction
+    public static int mapSize(String jsonString) {
+        // handle empty string case
+        if (jsonString == null) {
+            return 0;
+        }
+
+        // convert string input to equivalent java map
+        Map convertedMap = stringToMap(jsonString);
+
+        // handle both cases for where returned map is null or is valid but empty
+        if (convertedMap == null || convertedMap.entrySet().isEmpty()) {
+            return 0;
+        }
+
+        // return number of entries extracted
+        return convertedMap.entrySet().size();
+    }
+
+    /**
+     * Internal Testing Function getValueFromKey that returns given key's corresponding value for
+     * given JSON Map Object
+     *
+     * @param object
+     * @param key
+     * @return corresponding value for given key from JSON map in data
+     */
+    public static Object getValueFromKey(Object object, Object key) {
```

Comment:
This shouldn't be a public method in a non-testing class if it's only being used in unit tests.